src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -537,22 +537,9 @@ def setJobResourceInformation(self, timePerEvent=None, sizePerEvent=None, memory
             performanceParams.timePerEvent = timePerEvent or getattr(performanceParams, "timePerEvent")
         if sizePerEvent or getattr(performanceParams, "sizePerEvent", None):
             performanceParams.sizePerEvent = sizePerEvent or getattr(performanceParams, "sizePerEvent")
-
-        # special handling for memory overwrite during assignment
-        if isinstance(memoryReq, dict):
-            taskMemory = memoryReq.get(self.name())
-        else:
-            taskMemory = memoryReq
-
```

Comment:
Ideally not. The way I see it is that the Workload object could see a dictionary value, but it then passes a string to the task object. However, I think if we clone a taskChain it could be that setupProcessingTask
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/StdSpecs/StdBase.py#L336

would be handling a dictionary value.
I'll add this task.name() back here for all three arguments, but I think there is no need to call it for all the children from here.